OpenRA.Mods.Common/Widgets/Logic/Settings/HotkeysSettingsLogic.cs

```diff
@@ -228,17 +232,21 @@ void InitHotkeyRemapDialog(Widget panel)
 			var originalNoticeText = new CachedTransform<HotkeyDefinition, string>(hd => originalNotice.Text.F(hd?.Default.DisplayString()));
 			originalNotice.GetText = () => originalNoticeText.Update(selectedHotkeyDefinition);
 
+			var readonlyNotice = panel.Get<LabelWidget>("READONLY_NOTICE");
+			readonlyNotice.TextColor = ChromeMetrics.Get<Color>("NoticeInfoColor");
+			readonlyNotice.IsVisible = () => selectedHotkeyDefinition.Readonly;
+
 			var resetButton = panel.Get<ButtonWidget>("RESET_HOTKEY_BUTTON");
-			resetButton.IsDisabled = () => isHotkeyDefault;
+			resetButton.IsDisabled = () => isHotkeyDefault || selectedHotkeyDefinition.Readonly;
 			resetButton.OnClick = ResetHotkey;
 
 			var clearButton = panel.Get<ButtonWidget>("CLEAR_HOTKEY_BUTTON");
-			clearButton.IsDisabled = () => !hotkeyEntryWidget.Key.IsValid();
+			clearButton.IsDisabled = () => !hotkeyEntryWidget.Key.IsValid() || selectedHotkeyDefinition.Readonly;
```

Comment:
Check boolean values before calling functions.
```suggestion
			clearButton.IsDisabled = () => selectedHotkeyDefinition.Readonly || !hotkeyEntryWidget.Key.IsValid();
```